exasol/ai/mcp/server/connection_factory.py

```diff
@@ -0,0 +1,269 @@
+
+DEFAULT_CONN_POOL_SIZE = 5
+
+
+def local_env_complete(env: dict[str, Any]) -> bool:
+    return all(v in env for v in [ENV_DSN, ENV_USER]) and any(
+        v in env for v in [ENV_PASSWORD, ENV_ACCESS_TOKEN, ENV_REFRESH_TOKEN]
+    )
+
+
+def oidc_env_complete(env: dict[str, Any]) -> bool:
+    return all(v in env for v in [ENV_DSN, ENV_USERNAME_CLAIM])
+
+
+def saas_env_complete(env: dict[str, Any]) -> bool:
+    return (
+        all(v in env for v in [ENV_SAAS_HOST, ENV_SAAS_ACCOUNT_ID])
+        and any(v in env for v in [ENV_SAAS_PAT, ENV_SAAS_PAT_HEADER])
+        and any(v in env for v in [ENV_SAAS_DATABASE_ID, ENV_SAAS_DATABASE_NAME])
+    )
+
+
+def _copy_kwargs(env: dict[str, Any], name_map: dict[str, str]) -> dict[str, Any]:
+    return {val: env[key] for key, val in name_map.items() if key in env}
+
+
+def get_local_kwargs(env: dict[str, Any]) -> dict[str, Any]:
+    """
+    Returns pyexasol.connect(...) arguments in case of On-Prem pre-configured
+    server's credentials.
+    """
+    return _copy_kwargs(
+        env,
+        {
+            ENV_DSN: "dsn",
+            ENV_USER: "user",
+            ENV_PASSWORD: "password",
+            ENV_ACCESS_TOKEN: "access_token",
+            ENV_REFRESH_TOKEN: "refresh_token",
+        },
+    )
+
+
+def get_saas_kwargs(env: dict[str, Any]) -> dict[str, Any]:
+    """
+    Returns pyexasol.connect(...) arguments in case of SaaS backend.
+    If PAT is not pre-configured, the function will attempt to extract it from the
+    headers. The returned arguments is the result of an OpenAPI call that resolves
+    the SaaS username and password from the provided account id and the database id
+    or name.
+    """
+    saas_params = _copy_kwargs(
+        env,
+        {
+            ENV_SAAS_HOST: "host",
+            ENV_SAAS_ACCOUNT_ID: "account_id",
+            ENV_SAAS_PAT: "pat",
+            ENV_SAAS_DATABASE_ID: "database_id",
+            ENV_SAAS_DATABASE_NAME: "database_name",
+        },
+    )
+    if "pat" not in saas_params:
+        headers = fmcp_api.get_http_headers()
+        if env[ENV_SAAS_PAT_HEADER] not in headers:
+            raise RuntimeError(
+                "The SAAS PAT is not specified and cannot be extracted from the call headers."
+            )
+        saas_params["pat"] = headers[env[ENV_SAAS_PAT_HEADER]]
+    return saas_api.get_connection_params(**saas_params)
+
+
+def get_common_kwargs(env: dict[str, Any]) -> dict[str, Any]:
+    # TODO: Add SSL parameters.
+    return {
+        "fetch_dict": True,
+        "compression": True,
+    }
+
+
+def _create_connection_pool(
+    env: dict[str:Any],
+) -> NamedObjectPool[pyexasol.ExaConnection]:
+    pool_size = int(env.get(ENV_POOL_SIZE, DEFAULT_CONN_POOL_SIZE))
+    return NamedObjectPool(capacity=pool_size, cleanup=lambda conn: conn.close())
+
+
+def _build_impersonate_query(user: str) -> str:
+    # I can't figure out how to construct this query properly in SQLGlot
```

Comment:
Not possible, not supported in sqlglot. So, probably change the comment so that it is not supported in sqlglot.